visualization/app/codeCharta/ui/ribbonBar/colorMetricChooser/colorMetricChooser.component.ts

```diff
@@ -16,9 +19,14 @@ export class ColorMetricChooserComponent {
 	nonDisabledColor = "rgba(0, 0, 0, 0.38)"
 	disabledColor = "rgba(68,68,68, 1)"
 
-	constructor(private store: Store<CcState>) {}
+	constructor(private store: Store<CcState>, private state: State<CcState>) {}
 
 	handleColorMetricChanged(value: string) {
 		this.store.dispatch(setColorMetric({ value }))
 	}
+
+	resetColorRange = () => {
+		const selectedColorMetricData = selectedColorMetricDataSelector(this.state.getValue())
```

Comment:
we could add this side effect into `app\codeCharta\state\store\dynamicSettings\colorRange\resetColorRange.effect.ts` through adding `setColorMetric` to the action pipe

Reply:
@shaman-apprentice good idea! I tried to change it.